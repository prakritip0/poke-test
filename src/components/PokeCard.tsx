import React from 'react';
import {useFetch} from '../hooks/useFetch';
import {Link} from 'react-router-dom'

interface PokeCardPropsType {
  id: number;
}

const PokeCard = ({id}: PokeCardPropsType) => {
  const {data} = useFetch(
    `https://pokeapi.co/api/v2/pokemon/${id}/`
  );


  return (
    <Link to='/pokeInfo'>
    <div className='h-[5rem] w-[15rem] bg-red-300 border rounded-lg px-3 py-2 cursor-pointer' >
      {/* <div className='h-[6rem] w-[6rem] rounded-full overflow-hidden'>
        <img
          className='w-[100%] h-[100%]'
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`}
          alt={data.name}
        />
      </div> */}
      <h2 className='text-white text-lg font-bold'>
        {data?.name[0]?.toUpperCase() + data?.name?.slice(1)}
      </h2>
      <p className='text-white'>Id: {id}</p>
    </div>
    </Link>
  );
};

export default PokeCard;
