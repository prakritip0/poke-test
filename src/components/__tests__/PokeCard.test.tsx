import {fireEvent, render, screen} from '@testing-library/react';
import PokeCard from '../PokeCard';
import React from 'react';
import '@testing-library/jest-dom';
import {BrowserRouter} from 'react-router-dom';

describe('Pokedex home', () => {
  const mockUseNavigate = jest.fn();

  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate,
  }));

  const TestComponent = (
    <BrowserRouter>
      <PokeCard name='Bulbasaur' routeName='/details/bulbasaur' />
    </BrowserRouter>
  );
  it('Renders properly', () => {
    render(TestComponent);
    expect(screen.getByText('Bulbasaur')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const {baseElement} = render(TestComponent);
    expect(baseElement).toMatchSnapshot();
  });

  it('navigates to detail page', async () => {
    render(TestComponent);

    // fireEvent.click(await screen.findByTestId('pokemon-card'));

    // expect(mockUseNavigate).toBeCalled();
  });
});
