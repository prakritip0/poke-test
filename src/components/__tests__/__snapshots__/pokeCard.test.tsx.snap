// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokedex home matches snapshot 1`] = `
<body>
  <div>
    <a
      data-testid="pokemon-card"
      href="/details//details/bulbasaur"
    >
      <div
        class="bg-slate-100 px-3 py-5 rounded"
      >
        <p>
          Bulbasaur
        </p>
      </div>
    </a>
  </div>
</body>
`;
